here-naksha-lib-psql/src/commonMain/kotlin/naksha/psql/executors/write/UpdateCollection.kt

```diff
@@ -0,0 +1,70 @@
+package naksha.psql.executors.write
+
+import naksha.model.*
+import naksha.model.Naksha.NakshaCompanion.VIRT_COLLECTIONS_QUOTED
+import naksha.model.NakshaError.NakshaErrorCompanion.ILLEGAL_ARGUMENT
+import naksha.model.objects.NakshaCollection
+import naksha.model.objects.NakshaFeature
+import naksha.psql.*
+import naksha.psql.executors.WriteExt
+import naksha.psql.executors.write.WriteCollectionUtils.tupleOfCollection
+import naksha.psql.executors.write.WriteFeatureUtils.allColumnValues
+import naksha.psql.executors.write.WriteFeatureUtils.newFeatureTupleNumber
+
+class UpdateCollection(
+    private val session: PgSession
+) {
+
+    fun execute(map: PgMap, write: WriteExt): Tuple? {
+        // Note: write.collectionId is always naksha~collections!
+        val feature = write.feature?.proxy(NakshaCollection::class) ?: throw NakshaException(
+            NakshaError.ILLEGAL_ARGUMENT,
+            "UPDATE without collection as feature"
+        )
+        require(write.featureId != null) {
+            "Collection id not given"
+        }
```

Comment:
Please do not throw arbitrary exceptions, always throw `NakshaException`, in this case, this check anyway is a duplicate of the one below, please remove it.